pinot-spi/src/main/java/org/apache/pinot/spi/stream/Checkpoint.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.stream;
+
+/**
+ * Keeps track of the consumption for a PartitionGroup
```

Comment:
I still don't understand why we need this class. Why not just extend StreamPartitionOffset interface to include a fromString() method? (we already have a toString()).

I am also not opposed to adding two new methods serialize() and deserialize() if that somehow (I don't see how) makes things look/feel better. We can deprecate the existing toString() method.

Reply:
@npawar I also feel the same now that we should stick to a single class. Was there a specific motivation behind this change? If not, I'll make the changes suggested by @mcvsubbu

Reply:
Alright, lets not add Checkpoint.